pandas/core/arrays/base.py

```diff
@@ -867,6 +867,24 @@ def copy(self) -> ABCExtensionArray:
         """
         raise AbstractMethodError(self)
 
+    def view(self, dtype=None) -> ABCExtensionArray:
+        """
+        Return a view on the array.
+
+        Returns
+        -------
+        ExtensionArray
+
+        Notes
```

Comment:
I've been inconsistent about this, be in general I've tried to keep docstrings user-facing. For implementation notes I've just used regular comments.

What are the consequences of `.view` returning self?

Do we have any restrictions on this being zero-copy?

Reply:
> Do we have any restrictions on this being zero-copy?

The test added in this PR will fail if we don't get an actual view.

> What are the consequences of `.view` returning self?

`view` is going to be used by the default implementation of `reshape`, so returning self would cause all kinds of trouble.

The default implementation should Just Work as long as `self[:]` returns a view, which should be the case anyway (JSONArray ATM returns a _copy_)

Reply:
Convert this to normal comments as Tom mentioned?